### PR TITLE
Compute mode speedfix (not done)

### PR DESCRIPTION
**Pull request by giuseppec**

### CODE REVIEW

File: R/computeMode3.R [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  if (is.integer(x)) {
+    shift = -min(x) + 1L
+    if (shift > 0)
+      x = x + shift
+    mode = .Call("c_compute_mode", x, tm, PACKAGE = "BBmisc")
+    if (shift > 0)
+      mode = mode - shift
+    return(mode)
+  } else if (is.numeric(x) || is.character(x)) {
+    f = as.factor(x) # this slows everything down
```
Comment (giuseppec) [inline R/computeMode3.R]: horribly slow

File: R/computeMode3.R [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  if (is.integer(x)) {
+    shift = -min(x) + 1L
+    if (shift > 0)
+      x = x + shift
+    mode = .Call("c_compute_mode", x, tm, PACKAGE = "BBmisc")
+    if (shift > 0)
+      mode = mode - shift
+    return(mode)
+  } else if (is.numeric(x) || is.character(x)) {
```
Comment (giuseppec) [inline R/computeMode3.R]: add `is.factor(x)` here otherwise computeMode3 won't work for factors

File: R/computeMode3.R (hunk 1 above)
Comment (mllg) [reply R/computeMode3.R]: Handling non-integer variables will require a full-featured hash table implementation in C, or in other words, some hundred lines of code to get it right.

**Comment by berndbischl** [comment]
just leave this foir now. this is NOT urgent

**Comment by berndbischl** [comment]
the fix is to do this via sorting

**Comment by berndbischl** [comment]
> Handling non-integer variables will require a full-featured hash table implementation in C, or in other 
> words, some hundred lines of code to get it right.

it would be possible to simply copy a hashtable c implementation to BBmisc, that would be feasible

**Comment by berndbischl** [comment]
it is also very much possible to simply sort the vector, then compute the mode. thats O(n log n) and still ok. 
C code for this is in todo-files now. 

but: for easy maintenance i opted to simply use data.table

new PR is #82

**Pull request closed without merging**